client/src/templates/Challenges/classic/editor.tsx

```diff
@@ -608,14 +610,23 @@ const Editor = (props: EditorProps): JSX.Element => {
   const ariaAlert = (message: string) => {
     const ariaLive: NodeListOf<HTMLDivElement> =
       document.querySelectorAll('.monaco-alert');
+    const time = `t${Date.now()}`;
     if (ariaLive.length > 0) {
       const liveText = ariaLive[0];
+      liveText.dataset.timestamp = time;
       liveText.textContent = message;
       // Hack used by monaco to force older browsers to announce the update to
       // the live region.
       // See https://www.tpgi.com/html5-accessibility-chops-aria-rolealert-browser-support/
       liveText.style.visibility = 'hidden';
       liveText.style.visibility = 'visible';
+      // Need to remove message after a few seconds so screen readers don't
+      // run into it.
+      setTimeout(function () {
+        if (liveText.dataset.timestamp === time) {
+          liveText.textContent = '';
+        }
+      }, 3000);
```

Comment:
I don't quite understand this. Before these changes the `textContext` remains indefinitely. With these changes, the `textContext` always gets set to `''` after 3 seconds. What purpose does the timestamp serve? I would have thought

```
      setTimeout(() => (liveText.textContent = ''), 3000);
```
is all you need.

Reply:
This is to prevent a race condition. Consider that the text has been in the live region for just under 3 seconds and then the user hits `Ctrl+m` again which triggers a new message to the live region. If that new message is placed in the live region just before the previous message's 3 seconds is up then the the previous message's 3 second timer will delete the new message once it finally reaches 3 seconds. If there isn't enough time for the screen reader to detect the new message before it is deleted then the user will never hear the message. Attaching a unique id to each message and then only deleting the message if that unique id is still there prevents previous timers from erasing new messages.

Reply:
Thanks for clarifying, that makes sense now. Thing is, this feels pretty weird, since `setTimeout` already returns a unique id that can be used to cancel it. Using dataset is a clever solution, for sure, but the intention is somewhat obscured.

I'd recommend storing the id in a ref and simply cancelling the old timer before creating a new one.

Either that or create a helper function `setRemoveMessageTimeout` or similar. That way the relevant code is all in one place and it should be more obvious what's going on.

Reply:
Hey @ojeytonwilliams Thanks for the feedback. I think the reason I did it this way is because it feels weird to me to have the Editor class get involved in storing data for this functionality. The way it is now, everything needed to make this work is contained in the ariaAlert function and Editor doesn't have to know anything about it (other than the name of the function). I guess that's my OOP background coming into play. Would a few more comments explaining what is going on make this clearer? Otherwise, I will be happy to make the change you suggested (using a class variable to store the id and then cancelling the timer).

Reply:
> Editor doesn't have to know anything about it (other than the name of the function)

Yeah, that's a good point - the editor is messy enough as it is!

I don't want to discuss this to death, so I just went with a little re-arrangement (to keep all the related code close together) and a few explanatory comments. If you're happy with them, I'm happy to merge.

Reply:
Your changes make sense. Merge away!